boot/zephyr/include/io/io.h

```diff
@@ -67,6 +67,45 @@ static inline bool io_boot_skip_serial_recovery()
 {
     uint32_t rr = nrfx_reset_reason_get();
 
+#ifdef NRF_RESETINFO
+    /* The following reset reasons should allow to enter firmware recovery or
+     * loader through an IO state.
+     */
+    const uint32_t allowed_reset_reasons = (
+        /* Reset from power on reset (reset reason POR or BOR). */
+        NRFX_RESET_REASON_POR_MASK
+#ifdef RESETINFO_RESETREAS_GLOBAL_RESETPOR_Msk
+        /* Reset from power on reset (reset reason other than POR or BOR). */
+        | RESETINFO_RESETREAS_GLOBAL_RESETPOR_Msk
+#endif
+        /* Reset from pin reset is not masked: it always enables the io-based recovery. */
+        /* Reset from the watchdog timer. */
+        | NRFX_RESET_REASON_DOG_MASK
+        /* Reset from CTRL-AP. */
+        | NRFX_RESET_REASON_CTRLAP_MASK
+        /* Reset due to secure domain system reset request. */
+        | NRFX_RESET_REASON_SREQ_MASK
+        /* Reset due to secure domain watchdog 0 timer. */
+        | NRFX_RESET_REASON_SECWDT0_MASK
+        /* Reset due to secure domain watchdog 1 timer. */
+        | NRFX_RESET_REASON_SECWDT1_MASK
+        /* Reset due to secure domain lockup. */
```

Comment:
not sure what all these are defined on but will this file compile e.g. for nrf51 and nrf52 devices which I assume don't have these?

Reply:
To my understanding - the NRF_RESETINFO is secific to nRF54H and nRF92 families.
I tried to follow the definitions from here: https://github.com/zephyrproject-rtos/hal_nordic/blob/979a58d0829108b57f10e90b6c3fc35ab6206e4b/nrfx/helpers/nrfx_reset_reason.h#L225

The main issue it that on those SoCs either `NRFX_RESET_REASON_POR_MASK` or `RESETINFO_RESETREAS_GLOBAL_RESETPOR_Msk` is set, making it impossible to pass the `0` test.